### PR TITLE
LOOP-3830 PumpManager sync updates

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-3830

### CODE REVIEW

File: Loop/View Models/BolusEntryViewModel.swift
Comment (nhamming) [inline Loop/View Models/BolusEntryViewModel.swift]: [nit] this and similar code intended to stay?

**Review by nhamming — APPROVED** [review]
LGTM

File: Loop/View Models/BolusEntryViewModel.swift
Comment (ps2) [reply Loop/View Models/BolusEntryViewModel.swift]: No, thanks; will clean up.

File: Loop/Managers/DeviceDataManager.swift
Comment (rickpasetto) [inline Loop/Managers/DeviceDataManager.swift:437]: Unit test for this new code?

**Review by rickpasetto — APPROVED** [review]
LGTM, with one request to please add a unit test for the new code in `DeviceDataManager`

File: Loop/Managers/DeviceDataManager.swift
Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift:437]: Generally when you've asked for tests at the top layer managers, I've split it out into a separate object to be testable. I'm not sure that's a good idea here; the logic is fairly simple, and an abstraction that makes things clearer and not more complicated isn't obvious to me. This also feels borderline on the cost/value scale. Are you ok with letting this one slide?

Comment (rickpasetto) [reply Loop/Managers/DeviceDataManager.swift:437]: Seems like important logic, though, doesn't it?  I hear you that it is difficult to test right now.  Are there other ways to get this under test?

Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift:437]: I'm not just saying the cost is high. I'm also saying the value is low (not a situation where the test will serve as documentation, simple enough that it's not likely to be misunderstood, not a situation where testing will improve the composability or interface; will actually probably do the opposite, for a piece that is intentionally not reusable). But if you insist, I'll look for ways to do it.

Comment (ps2) [reply Loop/Managers/DeviceDataManager.swift:437]: So, looking into this more, I think our options are to 1) Refactor DeviceDataManager's collaborators (the other managers) into protocols, so we can create mocks for them and put DeviceDataManager under test. 2) Abstract out some kind of class for this logic to decide whether to loop or not, based on glucose storage success/fail, and pumpdata freshness. A "ToLoopOrNotDecider" class, if you will. Hooking it into glucose storage, and/or results thereof, and pumpMananager freshness or ability to trigger pump refresh, and the ability to trigger a Loop mean a lot more wiring that complicates a user understanding the flow.  All for the purpose of putting just two actual pieces of logic under test; the freshness comparison, and glucose storage success comparison. 

I don't think either option is a net improvement in the readability/simplicity of the code, so I am loathe to do it. Those options are usually good ideas for code that might be reusable or substitutable; but this thin layer of mostly declarative code in the top level Loop Managers isn't like that. Yes, it *is* important logic, and because of that, I think it's important to keep it simple. Unless you strongly object; and I'm guessing since you approved the PR, that you don't, I don't think we should do either of the options above.

Comment (rickpasetto) [reply Loop/Managers/DeviceDataManager.swift:437]: I won't insist, I was just asking as part of the code review.  I wouldn't have approved this if I did not leave it up to you to take into consideration.  I was concerned from a risk point of view; is this something that would benefit from being under automated testing?  In my opinion, the answer is yes.  
Would you be open to filing a separate LOOP task to address this later?  I would be fine with that.

**Pull request merged**
